secret-manager/cleaner.sh

```diff
@@ -136,18 +225,92 @@ else
   exit 1
 fi
 
-response=$(kubectl delete sa/xkops-secret-sa --namespace xkops 2>&1)
+#https://awscli.amazonaws.com/v2/documentation/api/latest/reference/cloudformation/describe-stacks.html
+response=$(aws cloudformation describe-stacks --stack-name eksctl-xkops-cluster-addon-iamserviceaccount-xkops-xkops-secret-sa 2>&1)
 # possible responses:
-# serviceaccount "xkops-secret-sa" deleted
-# Error from server (NotFound): serviceaccounts "xkops-secret-sa" not found
+# The output shows the stack information but we only need to check the stack name
+# "StackName": "eksctl-xkops-cluster-addon-iamserviceaccount-xkops-xkops-secret-sa"
+# An error occurred (ValidationError) when calling the DescribeStacks operation: Stack with id eksctl-xkops-cluster-addon-iamserviceaccount-xkops-xkops-secret-sa does not exist
 
-if [[ $response == *"deleted"* ]]; then
-  echo "Successfully deleted xkops-secret-sa service account."
-elif [[ $response == *"Error from server (NotFound)"*  ]]; then
-  echo "xkops-secret-sa service account has already been removed."
+if [[ $response == *'"StackName": "eksctl-xkops-cluster-addon-iamserviceaccount-xkops-xkops-secret-sa"'* ]]; then
+  # Refernce : https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/using-cfn-cli-deleting-stack.html
+  aws cloudformation delete-stack --stack-name eksctl-xkops-cluster-addon-iamserviceaccount-xkops-xkops-secret-sa
+  sleep 3
+  response2=$(aws cloudformation describe-stacks --stack-name eksctl-xkops-cluster-addon-iamserviceaccount-xkops-xkops-secret-sa 2>&1)
+  if [[ $response2 == *"does not exist"* ]]; then
+    echo "Successfully deleted xkops-secret-sa stack."
+  fi
+elif [[ $response == *"does not exist"*  ]]; then
+  echo "xkops-secret-sa stack has already been removed."
+else
+  echo "$response"
+  echo "Failed to remove xkops-secret-sa stack. Exiting..."
+  exit 1
+fi
+
+# Refernce : https://docs.aws.amazon.com/eks/latest/userguide/managing-add-ons.html
+response=$(eksctl delete addon --cluster xkops-cluster --name aws-ebs-csi-driver 2>&1)
+# possible responses:
+# deleted addon: aws-ebs-csi-driver
+
+if [[ $response == *"deleted addon:"* ]]; then
+  echo "Successfully deleted aws-ebs-csi-driver addon."
+elif [[ $response == *"does not exist"*  ]]; then
+  echo "aws-ebs-csi-driver addon has already been removed."
+else
+  echo "$response"
+  echo "Failed to remove aws-ebs-csi-driver addon. Exiting..."
+  exit 1
+fi
+
+
+# https://awscli.amazonaws.com/v2/documentation/api/latest/reference/iam/list-attached-role-policies.html
+response=$(aws iam list-attached-role-policies --role-name XkOps-EBS-iam-role 2>&1)
+# possible responses:
+# The response shows the detailes of the policy, we only need the PolicyName to confirm it exists.
+# "PolicyName": "AmazonEBSCSIDriverPolicy"
+# "AttachedPolicies": []
+
```

Comment:
This response needs to be handled 
`An error occurred (NoSuchEntity) when calling the ListAttachedRolePolicies operation: The role with name XkOps-EBS-iam-role cannot be found.`